src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(first, value, null);
+        if (isEmpty()) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
+            throw new IndexOutOfBoundsException("Incorrect index " + index + "; size is " + size());
+        }
+
+        if (index == size()) {
+            add(value);
+        } else {
+            Node<T> newNode = new Node<>(null, value, null);
+            Node<T> currentNode = findNodeByIndex(index);
+
+            if (index == 0) {
+                newNode.next = first;
+                first.prev = newNode;
+                first = newNode;
+            } else {
+                newNode.next = currentNode;
+                newNode.prev = currentNode.prev;
+                currentNode.prev.next = newNode;
+                currentNode.prev = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findNodeByIndex(index);
```

Comment:
This is code repeating. You always check the index before getting node by index. What you should do is move the check logic INSIDE the get method.